### ISSUE TITLE
HttpRequestExtensions GetUri() does not handle a PathBase on the request

### ISSUE THREAD

**Issue by jorenby** [LtiLibrary/LtiLibrary#68]
This leads to `request.ParseLtiRequestAsync().GenerateSignature("secret")` not matching the signature on the request.

For example, if we wrap [the UseMvc() call here](https://github.com/andyfmiller/LtiLibrary/blob/master/test/LtiLibrary.AspNetCore.Tests/Startup.cs#L30) with an `app.Map()` like:

```
            app.Map("/basePath", a =>
            {
                a.UseMvc(routes =>
                {
                    routes.MapRoute(
                        "default",
                        "{controller=Home}/{action=Index}/{id?}");
                });
            });
```

GetUri() will return something like `http://localhost/ims/outcomes` instead of `http://localhost/basePath/ims/outcomes`.

**Comment by jorenby** [comment]
I think the solution is as simple as adding request.PathBase to [this line](https://github.com/andyfmiller/LtiLibrary/blob/master/src/LtiLibrary.AspNetCore/Extensions/HttpRequestExtensions.cs#L120) like so:
```
Path = request.PathBase + request.Path,
```

I'd be happy to make a pull request, but I'm not sure how best to handle the testing. The existing tests will continue working, but to exercise the change we'd need to add the app.Map() to Startup.cs and change a bunch of paths. Or we make the Map parameter "/ims" and remove that part from the Controller routes.

**Comment by andyfmiller** [comment]
Would you make a pull request? I've never used Map, but it looks intriguing. I'd love to take a look.

**Comment by andyfmiller** [comment]
I tried out your suggestions in your comments. By adding app.Map("/basepath") to Startup, in addition to the default, then only one set of tests needed to be updated and all the others continued to work. See PR #69.

What do you think?

**Comment by jorenby** [comment]
Thanks for getting on this so quickly! Looks like that'll do it. Didn't occur to me to just have two UseMvc()s with different routes.

I've only ever used Map() to add a base path part for every endpoint in a service, as well as diagnostic pages and Swagger, etc.

**Comment by andyfmiller** [comment]
I merged the fix into the 3.0.1 branch.
